drivers/vmwarefusion/fusion_darwin.go

```diff
@@ -535,6 +540,18 @@ func (d *Driver) getIPfromVmnetConfiguration(macaddr string) (string, error) {
 	return "", fmt.Errorf("IP not found for MAC %s in vmnet configuration files", macaddr)
 }
 
+func (d *Driver) getIPfromVmrun() (string, error) {
+	vmx := d.vmxPath()
+
+	ip := regexp.MustCompile(`(\d+\.\d+\.\d+\.\d+)`)
+	stdout, _, _ := vmrun("getGuestIPAddress", vmx)
```

Comment:
I don't think this works in VMWare headless mode - there appears to be a [bug](https://communities.vmware.com/t5/VMware-Fusion-Discussions/Using-vmrun-to-obtain-a-guests-IP-address-when-running-the-VM-in/td-p/2236978) that requires passing the `-wait` option to `getGuestIPAddress` otherwise it returns `Error: The VMware Tools are not running in the virtual machine`. 

Also see https://github.com/machine-drivers/docker-machine-driver-vmware/pull/31#discussion_r588061432